### ISSUE TITLE
Langchain python agent calculation is wrong!!!

### ISSUE THREAD

**Issue by cosmosanalytics** [hwchase17/langchain#4551]
### System Info

I am using colab.
![image](https://github.com/hwchase17/langchain/assets/100080151/d1058dfb-f65d-41c6-be8f-806c9aef26d1)

### Who can help?

@hwchase17
@vowelparrot

### Information

- [ ] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [X] Agents / Agent Executors
- [X] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

![image](https://github.com/hwchase17/langchain/assets/100080151/3292307a-04b8-4f92-8428-38b3e4655a35)

### Expected behavior

![image](https://github.com/hwchase17/langchain/assets/100080151/e3c42af8-9c72-48ab-8a63-d341225e4bf7)

**Comment by cosmosanalytics** [comment]
this is the code
from langchain.agents.agent_toolkits import create_python_agent
from langchain.agents import create_pandas_dataframe_agent
from langchain.tools.python.tool import PythonREPLTool
from langchain.llms import OpenAI
import pandas as pd
import numpy as np

create_python_agent(OpenAI(temperature=0), PythonREPLTool(), verbose=True).run('''
what is the average gdpPercap for United States  in data https://raw.githubusercontent.com/plotly/datasets/master/gapminderDataFiveYear.csv
''')

AND

import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/gapminderDataFiveYear.csv')
us_data = df[df['country'] == 'United States']
us_avg_gdpPercap = us_data['gdpPercap'].mean()
us_avg_gdpPercap

**Comment by vowelparrot** [comment]
In this case th emodel never sees any of the tool output because the PythonREPL tool only returns whatever the agent `print()`'s.

If you ask

```
what is the average gdpPercap for United States in data https://raw.githubusercontent.com/plotly/datasets/master/gapminderDataFiveYear.csv?
    You must print() the variable.
    ```
    
    It returns the correct answer for me.

**Comment by cosmosanalytics** [comment]
yes, now I get correct answer
![image](https://github.com/hwchase17/langchain/assets/100080151/8e4c7964-7553-4d4e-9d82-e42ae4831739)
thanks!

**Comment by dsantiago** [comment]
Yeah, the answer is not generally reliable... Here calculation is also wrong

```python
from langchain.llms import OpenAI
from langchain.agents import load_tools, initialize_agent, AgentType

llm = OpenAI(model_name="text-davinci-003", temperature=0)
tools = load_tools(["python_repl", "wikipedia"], llm=llm)

agent = initialize_agent(
		tools,
		llm,
		agent=AgentType.ZERO_SHOT_REACT_DESCRIPTION,
		verbose=True
	)

agent.run("Who is Olivia Wilde's boyfriend? Tell me his current age and his age raised to the 0.23 power?")
```

Output:
`Thought: I now know the final answer`
`Final Answer: Olivia Wilde's boyfriend is Harry Styles, who is currently 27 years old and his age raised to the 0.23 power is  approximately 5.`
